tests/dotnet/UnitTests/XcodeProjectTests.cs

```diff
@@ -0,0 +1,378 @@
+			var outputFxThirdWriteTime = File.GetLastWriteTime (expectedXcodeFxOutput);
+			Assert.IsTrue (outputFxThirdWriteTime > outputFxFirstWriteTime, $"Expected '{outputFxThirdWriteTime}' write time of '{outputFxThirdWriteTime}' to be greater than first write '{outputFxFirstWriteTime}'");
+		}
+
+		[Test]
+		public void BuildMultipleSchemesiOS ()
+		{
+			var platform = ApplePlatform.iOS;
+			Configuration.IgnoreIfIgnoredPlatform (platform);
+
+			var testDir = Cache.CreateTemporaryDirectory (TestName);
+			var proj = Path.Combine (testDir, $"{TestName}.csproj");
+			DotNet.AssertNew (testDir, "iosbinding");
+
+			var xcodeProjName = "TwoSchemeFx";
+			var xcodeProjDirSrc = Path.Combine (XCodeTestProjectDir, xcodeProjName);
+			var xcodeProjDirDest = Cache.CreateTemporaryDirectory ($"{TestName}XcProj");
+			FileHelpers.CopyDirectory (xcodeProjDirSrc, xcodeProjDirDest);
+			AddXcodeProjectItem (proj, Path.Combine (xcodeProjDirDest, $"{xcodeProjName}.xcodeproj"),
+				new Dictionary<string, string> {
+					{ "SchemeName", xcodeProjName },
+				});
+
+			var secondSchemeName = "iOSFx";
+			AddXcodeProjectItem (proj, Path.Combine (xcodeProjDirDest, $"{xcodeProjName}.xcodeproj"),
+				new Dictionary<string, string> {
+					{ "SchemeName", secondSchemeName },
+				});
+
+			DotNet.AssertBuild (proj);
+			AssertXcFrameworkOutput (platform, testDir, xcodeProjName);
+			AssertXcFrameworkOutput (platform, testDir, secondSchemeName);
+		}
+
+		[Test]
+		public void BuildMultipleProjectsiOS ()
+		{
+			var platform = ApplePlatform.iOS;
+			Configuration.IgnoreIfIgnoredPlatform (platform);
+
+			var testDir = Cache.CreateTemporaryDirectory (TestName);
+			var proj = Path.Combine (testDir, $"{TestName}.csproj");
+			DotNet.AssertNew (testDir, "iosbinding");
+
+			var xcodeProjName = "TemplateFx";
+			var xcodeProjDirSrc = Path.Combine (XCodeTestProjectDir, xcodeProjName);
+			var xcodeProjDirDest = Cache.CreateTemporaryDirectory ($"{TestName}XcProj");
+			FileHelpers.CopyDirectory (xcodeProjDirSrc, xcodeProjDirDest);
+			AddXcodeProjectItem (proj, Path.Combine (xcodeProjDirDest, $"{xcodeProjName}.xcodeproj"),
+				new Dictionary<string, string> {
+					{ "SchemeName", xcodeProjName },
+				});
+
+			var xcodeProjName2 = "TwoSchemeFx";
+			var xcodeProjDirSrc2 = Path.Combine (XCodeTestProjectDir, xcodeProjName2);
+			var xcodeProjDirDest2 = Cache.CreateTemporaryDirectory ($"{TestName}XcProjTwo");
+			FileHelpers.CopyDirectory (xcodeProjDirSrc2, xcodeProjDirDest2);
+			AddXcodeProjectItem (proj, Path.Combine (xcodeProjDirDest2, $"{xcodeProjName2}.xcodeproj"),
+				new Dictionary<string, string> {
+					{ "SchemeName", xcodeProjName2 },
+				});
+
+			DotNet.AssertBuild (proj);
+			AssertXcFrameworkOutput (platform, testDir, xcodeProjName);
+			AssertXcFrameworkOutput (platform, testDir, xcodeProjName2);
+		}
+
+		[Test]
+		public void BuildMultiTargeting ()
+		{
+			var testDir = Cache.CreateTemporaryDirectory (TestName);
```

Comment:
If you want the test to run when all platforms are available, you can do this:

```cs
Configuration.IgnoreIfAnyIgnoredPlatforms ();
```

otherwise you need to assert that at least iOS is available:

```cs
Configuration.IgnoreIfIgnoredPlatform (ApplePlatform.iOS);
```

(although the test seems to need at least another platform, and you'll have to pick one if you don't want to just require all of them).

Reply:
I figured this would safely run against all enabled platforms, and that it would be pretty rare for a branch to exist that only supports one platform? Even in that case it's probably fine for this to still run and not truly test multitargeting, as there should be a corresponding branch that will have multiple platforms enabled be testing a multitargeted project?

Reply:
For all branches, we run these tests concurrently in 5 modes (in 5 different bots; this is to speed up the test run):

1. iOS enabled, all other platforms disabled + excluding tests with the "Multiplatform" category.
2. tvOS enabled, all other platforms disabled + excluding tests with the "Multiplatform" category.
3. macOS enabled, all other platforms disabled + excluding tests with the "Multiplatform" category.
4. Mac Catalyst enabled, all other platforms disabled + excluding tests with the "Multiplatform" category.
5. All available platforms enabled, only tests with the "Multiplatform" category.

So while it's technically really rare that a branch exists that only supports one platform, we run these tests with only one platform all the time.

It's not as rare to have a branch where iOS is completely disabled (we've done macOS/Mac Catalyst-only release branches in the past), in which case the multi platform variation will fail as it stands, because iOS won't be enabled for creating a a new "iosbinding" project.

This is why you need either of the `Configuration.IgnoreIf[...]` statements I mentioned.

Reply:
Thanks for the context, I'll make a couple of changes here to check for an enabled platform to create the template.